src/main/java/mate/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,50 @@
+package mate.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls = new HashSet<>();
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+        allowedUrls.add("/");
+        allowedUrls.add("/index");
```

Comment:
The user should only have access to two requests. /drivers/create and /login.

Reply:
Your index page display only the menu so it's quite ok to have access to it but when the user choose some item he is redirected to login menu

Reply:
Yes, you're right.